moped-editor/src/views/projects/projectsListView/ProjectsListViewTableToolbar.js

```diff
@@ -0,0 +1,140 @@
+  const [columnsButtonAnchorEl, setColumnsButtonAnchorEl] = useState(null);
+  const { classes } = props;
+  const localization = {
+    ...ProjectsListViewTableToolbar.defaultProps.localization,
+    ...props.localization,
+  };
+
+  return (
+    <Toolbar ref={props.forwardedRef} className={classes.root}>
+      <div className={classes.spacer} />
+      <div className={classes.actions}>
+        <div>
+          <div style={{ display: "flex" }}>
+            {props.columnsButton && (
+              <span>
+                <Tooltip title={localization.showColumnsTitle}>
+                  <IconButton
+                    color="inherit"
+                    onClick={event =>
+                      setColumnsButtonAnchorEl(event.currentTarget)
+                    }
+                    aria-label={localization.showColumnsAriaLabel}
+                  >
+                    <props.icons.ViewColumn />
+                  </IconButton>
+                </Tooltip>
+                <Menu
+                  anchorEl={columnsButtonAnchorEl}
+                  open={Boolean(columnsButtonAnchorEl)}
+                  onClose={() => setColumnsButtonAnchorEl(null)}
+                >
+                  <MenuItem
+                    key={"text"}
+                    disabled
+                    style={{
+                      opacity: 1,
+                      fontWeight: 600,
+                      fontSize: 12,
+                    }}
+                  >
+                    {localization.addRemoveColumns}
+                  </MenuItem>
+
+                  {/**
+                   * Add columns to the Columns Button Menu
+                   * aka the menu that pops up when `MaterialTable.options.columnsButton` is true
+                   */}
+                  {props.columns.map(col => {
+                    const hiddenFromColumnsButtonMenu =
+                      col.hiddenByColumnsButton !== undefined
+                        ? col.hiddenByColumnsButton
+                        : props.columnsHiddenInColumnsButton;
+
+                    if (hiddenFromColumnsButtonMenu) {
+                      return null;
+                    }
+
+                    return (
+                      <li key={col.tableData.id}>
+                        <MenuItem
+                          className={classes.formControlLabel}
+                          component="label"
+                          htmlFor={`column-toggle-${col.tableData.id}`}
+                          disabled={col.removable === false}
+                        >
+                          <Checkbox
+                            checked={!col.hidden}
+                            id={`column-toggle-${col.tableData.id}`}
+                            onChange={() => {
+                              props.setHiddenColumns({
+                                ...props.columnConfiguration,
+                                [col.field]: !col.hidden,
+                              });
```

Comment:
Oh no, this is totally a regression. I was thinking it had to do with my changes to the body component, but its because of the function on line 83 -- I'll figure out how to prevent it from closing upon click